reactioncommerce/reaction
# Fix #4141: Options now have parent's taxCode

Resolves #4181   
Impact: **minor**  
Type: **bugfix**

## Issue
The `taxCode` field was only available on variant level, not on option level. This was fine for the UI, but even in the database, options were not having the `taxCode` field updated/created. Hence all the options had the default value of `0000` in the `taxCode` field.

## Solution
Whenever a product is published, the `taxCode` field from its variants is copied over to its options. Also whenever a option is created, it gets a initial value of `taxCode` same as of its parent.

## Breaking changes
None

## Testing
1. Create a product with variants and options.
2. Set a taxCode on the product and publish.
3. Look in the database for its options. They should have the updated `taxCode` field.
4. Create a new option for this product.
5. Look in the database for this option. It should have the `taxCode` same as that of parent.
